consensus/polybft/consensus_runtime.go

```diff
@@ -91,6 +91,8 @@ type consensusRuntime struct {
 	// state is reference to the struct which encapsulates bridge events persistence logic
 	state *State
 
+	fsm *fsm
```

Comment:
Add a comment here?

Reply:
Looking at the code of IBFT, it seems like we are going to access to the `fsm` from the multiple go routines:

```go
// Start the round timer worker
go i.startRoundTimer(ctxRound, currentRound)

//	Jump round on proposals from higher rounds
go i.watchForFutureProposal(ctxRound)

//	Jump round on certificates
go i.watchForRoundChangeCertificates(ctxRound)

// Start the state machine worker
go i.startRound(ctxRound)
```

Should we ensure thread safety for accessing it?

Reply:
We do not need, because `fsm` is created only before `runSequence` is executed (and all those routines access `fsm` after that). What we need is to make `fsm` immutable. Currently only `commitmentToSaveOnRegister` is problematic

Reply:
Comment has been added https://github.com/0xPolygon/polygon-edge/pull/865/commits/e9f82c8aa934344e1b727726a10daa473becae37